src/session.coffee

```diff
@@ -30,6 +30,8 @@ define [
           @_extractAnalyticsSessionOnCreate()
 
         connect: (shop_code)->
+          return console?.warn?('Connect called without a shop code') unless shop_code
```

Comment:
We could easily catch the whitespace cases as well: `skroutz_analytics('session', 'connect', '  ');`. Although I see no such requests in the logs.

```coffeescript
return console?.warn?('Connect called without a shop code') if !shop_code || shop_code.trim?() == ''
```

Reply:
If we decide to move forward with this we should add an extra test. Also we could add another test when the shop_code is null.

Reply:
I agree with this, we should trim the `shop_code` and then check it. We should update the specs to test for `null`, `''` and `' '`.

Reply:
I would be tempted to take this a step further and suggest a basic validation along the lines of `SA-[0-9]{4,}-[0-9]{4,}`

Reply:
Discussed offline. We decided to not trim the shop code.